recipes/kfserving.rb

```diff
@@ -0,0 +1,267 @@
+end
+
+kube_hops_kubectl 'apply-kfseving' do
+  user node['kube-hops']['user']
+  group node['kube-hops']['group']
+  url "#{node['kube-hops']['kfserving']['base_dir']}/kfserving.yaml"
+end
+
+# Hops system config
+
+directory "#{node['kube-hops']['hops-system']['base_dir']}" do
+  owner node['kube-hops']['user']
+  group node['kube-hops']['group']
+  mode '0700'
+  action :create
+end
+
+hopsworks_api_fqdn = consul_helper.get_service_fqdn("hopsworks.glassfish")
+hopsworks_api_port = 8181
+if node.attribute?('hopsworks') and node['hopsworks'].attribute?('https') and node['hopsworks']['https'].attribute?('port')
+  hopsworks_api_port = node['hopsworks']['https']['port']
+end
+
+template "#{node['kube-hops']['hops-system']['base_dir']}/hops-system.yaml" do
+  source "hops-system.yml.erb"
+  owner node['kube-hops']['user']
+  group node['kube-hops']['group']
+  variables({
+    'hopsworks_fqdn': hopsworks_api_fqdn,
+    'hopsworks_port': hopsworks_api_port
+  })
+end
+
+kube_hops_kubectl 'apply-hops-system' do
+  user node['kube-hops']['user']
+  group node['kube-hops']['group']
+  url "#{node['kube-hops']['hops-system']['base_dir']}/hops-system.yaml"
+end
+
+# Model Serving Admission Controller
+
+directory "#{node['kube-hops']['model-serving-webhook']['base_dir']}" do
+  owner node['kube-hops']['user']
+  group node['kube-hops']['group']
+  mode '0700'
+  action :create
+end
+
+template "#{node['kube-hops']['model-serving-webhook']['base_dir']}/model-serving-webhook.yaml.template" do
+  source "model-serving-webhook.yml.erb"
+  owner node['kube-hops']['user']
+  group node['kube-hops']['group']
+end
+
+kube_certs = "#{node['certs']['dir']}/kube/certs"
+kube_private = "#{node['certs']['dir']}/kube/private"
+
+bash 'configure-model-serving-webhook-tls' do
+  user 'root'
+  group 'root'
+  environment ({ 'HOME' => ::Dir.home(node['kube-hops']['user']),
+                 'KEYPW' => node['kube-hops']['pki']['ca_keypw'] })
+  cwd "#{node['kube-hops']['model-serving-webhook']['base_dir']}"
+  code <<-EOH
+    set -e
+
+    namespace="hops-system"
+    service_name="model-serving-webhook"
+    tmp_certs="$(mktemp -d)"
+    chmod 0700 "$tmp_certs"
+
+    # Generate the private key for the webhook server
+    [ -f ${tmp_certs}/${service_name}-tls.key.pem ] || openssl genrsa -out ${tmp_certs}/${service_name}-tls.key.pem 4096
+
+    # Generate a Certificate Signing Request (CSR) for the private key
+    [ -f ${tmp_certs}/${service_name}-tls.csr.pem ] || openssl req -new -key ${tmp_certs}/${service_name}-tls.key.pem -subj "/CN=$service_name.$namespace.svc" \
+      -out ${tmp_certs}/${service_name}-tls.csr.pem -passout pass:${KEYPW}
+
+    # Sign CSR
+    [ -f ${tmp_certs}/${service_name}-tls.crt.pem ] || openssl x509 -req -CA #{kube_certs}/kube-ca.cert.pem -CAkey #{kube_private}/kube-ca.key.pem -CAcreateserial \
+      -passin pass:${KEYPW} -in ${tmp_certs}/${service_name}-tls.csr.pem -out ${tmp_certs}/${service_name}-tls.crt.pem
```

Comment:
I thought we agreed on getting rid of this and sign the certificates using Hopsworks.

Reply:
done